rclpy/src/rclpy/publisher.cpp

```diff
@@ -143,6 +143,18 @@ Publisher::publish_raw(std::string msg)
   }
 }
 
+bool
+Publisher::wait_for_all_acked(rcl_duration_t pytimeout)
+{
+  rcl_ret_t ret = rcl_publisher_wait_for_all_acked(rcl_publisher_.get(), pytimeout.nanoseconds);
+  if (RCL_RET_OK == ret) {
+    return true;
+  } else if (RCL_RET_TIMEOUT == ret) {
+    return false;
+  }
+  throw RCLError("Failed to wait for all acknowledgements");
```

Comment:
shouldn't this use the rcl error message?

Reply:
Maybe using [append_rcl_error](https://github.com/ros2/rclpy/blob/a1a874742e76850c5a9e04c8392104e1e2e78121/rclpy/src/rclpy/exceptions.cpp#L34) (?).

Reply:
RCLError calls `append_rcl_error()` internally: https://github.com/ros2/rclpy/blob/a6aac563af6fa9825d8ea686ede6ad6df74d8a16/rclpy/src/rclpy/exceptions.cpp#L56

Reply:
ah, didn't see that.
Thanks!